### PR TITLE
Add missing includes and fix 2 function calls

### PR DESCRIPTION
**Pull request by matty45**
This fixes some missing includes that were causing compilation errors, also fixes a few function calls within studio.h

### CODE REVIEW

File: game/client/particlemgr.h [hunk 1]
```diff
@@ -119,9 +119,6 @@ entities. Each one is useful under different conditions.
 #include "tier0/fasttimer.h"
 #include "utllinkedlist.h"
 #include "utldict.h"
-#ifdef WIN32
-#include <typeinfo.h>
-#else
 #include <typeinfo>
 #endif
```
Comment (danielga) [inline game/client/particlemgr.h]: You forgot to remove the `#endif`.

File: public/studio.h [hunk 2]
```diff
@@ -1773,7 +1773,7 @@ struct thinModelVertices_t
 	{
 		Assert( ( m_numBoneInfluences  >= 1 ) && ( m_numBoneInfluences  <= 3 ) );
 		Assert( ( boneWeights.numbones >= 1 ) && ( boneWeights.numbones <= m_numBoneInfluences ) );
-		int    numStoredWeights = std::max( 0, ( m_numBoneInfluences - 1 ) );
+		int    numStoredWeights = max( 0, ( m_numBoneInfluences - 1 ) );
```
Comment (danielga) [inline public/studio.h:1776]: What error is this supposed to fix?

**Review by danielga — CHANGES_REQUESTED** [review]

File: game/client/particlemgr.h (hunk 1 above)
Comment (matty45) [reply game/client/particlemgr.h]: Sorry about that, will fix that later

File: public/studio.h (hunk 2 above)
Comment (matty45) [reply public/studio.h:1776]: I do not remember the exact error but I'll go and find it again. It was preventing compilation as far as I know.

Comment (matty45) [reply public/studio.h:1776]: hmm, thats odd, i cant seem to trigger the error again, ill just revert this commit unless it pops up again.

Comment (matty45) [reply public/studio.h:1776]: Screenshot of the errors

Comment (danielga) [reply public/studio.h:1776]: Can you revert these 2 lines then, please?

Comment (matty45) [reply public/studio.h:1776]: Error popped up again as soon as I reverted the changes.

**Review by danielga — APPROVED** [review]

**Pull request merged**